### DISCUSSION TITLE
Possible to run a plugin after parsing but before SWC transforms?

### DISCUSSION THREAD

**Discussion by dannymcgee** [swc-project/swc#4972] [Q&A] [answered]
I'm currently using some TS legacy decorators in a project to modify class prototypes, e.g. using `Object.defineProperty` to create accessors with some custom logic. However, I'd prefer to do this in a build step instead of having to invoke the decorators at runtime in the browser. I created a quick-and-dirty plugin and wired it up, but it seems like the decorators have already been compiled away by the time the plugin gets executed. Is there some configuration option I'm missing that would make this possible?

**Comment by kdy1** [comment, marked as answer]
Decorators are special and you can't run your plugin before `decorator`, `typescript::strip` nor `resolver`

**Reply by dannymcgee** [reply]
@kdy1 Thanks for the quick response! Would you be willing to elaborate on "decorators are special"? I,e, is there some technical constraint which would make it difficult to leave decorators intact for plugins to handle, or is it just due to the current design?

**Reply by kdy1** [reply]
Decorators pass should be invoked before invoking typescript strip pass.
But `resolver` and `typescript::strip` should be applied before invoking plugins, and changing it is a breaking change
